atc/engine/check_delegate.go

```diff
@@ -208,6 +210,14 @@ func (d *checkDelegate) PointToCheckedConfig(scope db.ResourceConfigScope) error
 	return nil
 }
 
+func (d *checkDelegate) UpdateScopeLastCheckStartTime(scope db.ResourceConfigScope) (bool, error) {
+	return scope.UpdateLastCheckStartTime(d.build.ID(), d.build.PublicPlan())
+}
+
+func (d *checkDelegate) UpdateScopeLastCheckEndTime(scope db.ResourceConfigScope, succeeded bool) (bool, error) {
+	return scope.UpdateLastCheckEndTime(succeeded)
```

Comment:
Trying to wrap my head around whether this could cause issues if we have two checks running "concurrently". Suppose we have two checks, `A` and `B`. Consider the following order of operations:

1. `A` calls `UpdateScopeLastCheckStartTime`, so the scope has `last_check_build_id: A`
2. `B` calls `UpdateScopeLastCheckStartTime`, so the scope has `last_check_build_id: B`
3. `A` calls `UpdateScopeLastCheckEndTime` with `succeeded = true`, so the scope has `last_check_succeeded: true`

Then, the build summary for the resource would report that:
1. The last check was for build B, and it succeeded
2. There is no build currently running (because the last check succeeded)

Reply:
That should not happen. With `checkDelegate.WaitToRun`, `A` and `B` will run in serial order.

Reply:
Ah whoops, not sure why but I thought for some reason this was called before `WaitToRun`